crates/shared/src/price_estimation/quasimodo.rs

```diff
@@ -86,11 +75,17 @@ impl QuasimodoPriceEstimator {
 
         let token_pair = TokenPair::new(query.sell_token, query.buy_token).unwrap();
         let pairs = self.base_tokens.relevant_pairs([token_pair].into_iter());
+        let gas_model = GasModel {
+            native_token: self.native_token,
+            gas_price: gas_price.to_f64_lossy(),
+        };
 
-        let amms = self
+        let balancer_pools = self.balancer_pools(pairs.clone(), &gas_model).await?;
+        let mut amms: BTreeMap<usize, AmmModel> = self
             .pools
             .fetch(pairs, Block::Recent)
```

Comment:
Maybe for symmetry we can have a `uniswaplike_pools()` with this original fetching code, then all the pools becomes `self.uniswaplike_pools().into_iter().chain(self.balancer_pools()).enumerate().collect()`.

Reply:
I did it this way to make the diff smaller. But agree for the future separate methods is better. Changed.